modin/pandas/groupby.py

```diff
@@ -850,27 +929,37 @@ def _wrap_aggregation(
         ----------
         qc_method : callable
             The query compiler method to call.
-        default_func : callable
-            The function to call if we need to default to pandas.
-        drop : bool, default: True
-            Whether or not to the grouping columns should be dropped on this operation.
         numeric_only : bool, default: True
             True for numeric only computations, False otherwise.
+        agg_args : list-like, optional
+            Positional arguments to pass to the aggregation function.
+        agg_kwargs : dict-like, optional
+            Keyword arguments to pass to the aggregation function.
         **kwargs : dict
-            The keyword arguments to be passed to the calling function.
+            Keyword arguments to pass to the specified query compiler's method.
 
         Returns
         -------
         DataFrame or Series
             Returns the same type as `self._df`.
         """
-        if self._axis != 0:
-            return self._default_to_pandas(default_func, **kwargs)
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if not self._is_multi_by and drop and self._drop and self._as_index:
-            groupby_qc = self._query_compiler.drop(columns=self._by.columns)
+        agg_args, agg_kwargs = map(
+            value_or_default, (agg_args, agg_kwargs), (tuple, dict)
+        )
+
+        if numeric_only and self.ndim == 2:
+            by_cols = self._internal_by
+            mask_cols = [
+                col
+                for col, dtype in self._df.dtypes.items()
+                if is_numeric_dtype(dtype)
+                or (
+                    isinstance(dtype, pandas.CategoricalDtype)
+                    and is_numeric_dtype(dtype.categories.dtype)
+                )
+                or col in by_cols
+            ]
+            groupby_qc = self._query_compiler.getitem_column_array(mask_cols)
```

Comment:
this will conflict with the same construction in #3298, have to align it with the new behaviour after #3298 will be merged